proposals/records.md

```diff
@@ -1,54 +1,60 @@
 
-# Records Work-in-Progress
+# Records
 
-Unlike the other records proposals, this is not a proposal in itself, but a work-in-progress designed to record consensus design
-decisions for the records feature. Specification detail will be added as necessary to resolve questions.
+This proposal tracks the specification for the C# 9 records feature, as agreed to by the C#
+language design team.
 
-The syntax for a record is proposed to be added as follows:
+The syntax for a record is as follows:
 
 ```antlr
-class_declaration
-    : attributes? class_modifiers? 'partial'? 'class' identifier type_parameter_list?
-      parameter_list? type_parameter_constraints_clauses? class_body
+record_declaration
+    : attributes? class_modifier* 'partial'? 'record' identifier type_parameter_list?
+      parameter_list? record_base? type_parameter_constraints_clause* record_body
     ;
 
-struct_declaration
-    : attributes? struct_modifiers? 'partial'? 'struct' identifier type_parameter_list?
-      parameter_list? struct_interfaces? type_parameter_constraints_clauses? struct_body
+record_base
+    : ':' class_type argument_list?
+    | ':' interface_type_list
+    | ':' class_type argument_list? interface_type_list
     ;
 
-class_body
-    : '{' class_member_declarations? '}'
-    | ';'
-    ;
-
-struct_body
-    : '{' struct_members_declarations? '}'
+record_body
+    : '{' class_member_declaration* '}'
     | ';'
     ;
 ```
 
-The `attributes` non-terminal will also permit a new contextual attribute, `data`.
-
-A class (struct) declared with a parameter list or `data` modifier is called a record class (record struct), either of which is a record type.
-
-It is an error to declare a record type without the `data` modifier.
+Record types are reference types, similar to a class declaration. It is an error for a record to provide
+a `record_base` `argument_list` if the `record_declaration` does not contain a `parameter_list`.
```

Comment:
@agocke could you explain the rationale behind this restriction? (related discussion: https://github.com/dotnet/csharplang/issues/3795)

Thanks!